EKF/python/ekf_derivation/generated/covariance_generated_compare.cpp

```diff
@@ -0,0 +1,972 @@
+#include <math.h>
+#include <stdio.h>
+#include <cstdlib>
+#include "../../../../matrix/matrix/math.hpp"
+
+typedef matrix::SquareMatrix<float, 24> SquareMatrix24f;
+
+inline float sq(float in) {
+    return in * in;
+}
+
+namespace ecl{
+	inline float powf(float x, int exp)
+	{
+		float ret;
+		if (exp > 0) {
+			ret = x;
+			for (int count = 1; count < exp; count++) {
+				ret *= x;
+			}
+			return ret;
+		} else if (exp < 0) {
+			ret = x;
+			for (int count = -1; count > exp; count--) {
+				ret *= x;
+			}
+			return 1.0f / ret;
+		} else {
+			return 1.0f;
+		}
+	}
+}
+
+int main()
+{
+    // test integer power function
+    const float x_in = 2.0f;
+    printf("ecl::powf(0.5,n) = %e , %e , %e , %e , %e\n",(double)ecl::powf(x_in,-2),
```

Comment:
It is good practice to do these checks of `ecl::powf` as unit tests. You could write them for example in a new `test/test_EKF_utils.cpp`. You could follow `test/test_AlphaFilter.cpp` for some guidance.